om4labs/diags/generic_section_transport/generic_section_transport.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+
+import argparse
+import io
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+from om4labs import m6plot
+import xarray as xr
+
+from om4labs.om4common import image_handler
+from om4labs.om4parser import default_diag_parser
+
+
+def parse(cliargs=None, template=False):
+    """
+    Function to capture the user-specified command line options
+    """
+    description = """ """
+
+    parser = default_diag_parser(description=description, template=template)
+
+    parser.add_argument(
+        "--passage_label",
+        type=str,
+        default="",
+        help="Text to use for the name of the passage being plotted",
+    )
+
+    parser.add_argument(
+        "--obsrange",
+        type=str,
+        default=None,
+        help="Comma-separated tuple of min and max observed ranges (e.g. -20,20)",
+    )
+
+    if template is True:
+        return parser.parse_args(None).__dict__
+    else:
+        return parser.parse_args(cliargs)
+
+
+def read(dictArgs, varlist=["umo", "vmo"]):
+    """Reads umo and vmo files from a working directory"""
+
+    infile = dictArgs["infile"]
+
```

Comment:
maybe idiotproof with:

```python

ds = xr.open_mfdataset(infile, use_cftime=True, combine="by_coords")
for var in varlist:
    if var not in ds.variables:
        raise ValueError(f"variable {var} not found in {dictArgs["infile"]}")

Reply:
Thanks, @raphaeldussin. I addressed this in f38464d.